AutomatedTesting/Gem/PythonTests/scripting/EditMenu_Default_UndoRedo.py

```diff
@@ -49,58 +53,55 @@ class EditMenu_Default_UndoRedo:
-        general.open_pane(SCRIPT_CANVAS_UI)
-        helper.wait_for_condition(lambda: general.is_pane_visible(SCRIPT_CANVAS_UI), WAIT_TIME_3)
-        scripting_tools.initialize_sc_editor_objects(self)
-
-        # 2) Open Variable Manager if not opened already
-        scripting_tools.initialize_variable_manager_object(self)
-
-        # 3) Create Graph
-        scripting_tools.create_new_sc_graph(self.sc_editor_main_window)
-
-        # 4) Create and verify the new variable exists in variable manager
-        scripting_tools.create_new_variable(self, VARIABLE_TYPES[0])
-        graph_variables = self.variable_manager.findChild(QtWidgets.QTableView, GRAPH_VARIABLES_QT)
-        row_count = 1
-        Report.result(Tests.variable_created, helper.wait_for_condition(
-            lambda: self.get_graph_variables_row_count(graph_variables) == row_count, WAIT_TIME_3))
-
-        # 5) Trigger Undo action and verify if variable is removed in Variable Manager
-        undo_redo_action = self.sc_editor.findChild(QtWidgets.QAction, "action_Undo")
-        undo_redo_action.trigger()
-        row_count = 0
-        Report.result(Tests.undo_worked, helper.wait_for_condition(
-            lambda: self.get_graph_variables_row_count(graph_variables) == row_count, WAIT_TIME_3))
-
-        # 6) Trigger Redo action and verify if variable is re-added in Variable Manager
-        undo_redo_action = self.sc_editor.findChild(QtWidgets.QAction, "action_Redo")
-        undo_redo_action.trigger()
-        row_count = 1
-        Report.result(Tests.redo_worked, helper.wait_for_condition(
-            lambda: self.get_graph_variables_row_count(graph_variables) == row_count, WAIT_TIME_3))
-
-        # 7) Close SC window
-        general.close_pane(SCRIPT_CANVAS_UI)
 
 
-test = EditMenu_Default_UndoRedo()
-test.run_test()
+    # Preconditions
+    from PySide2 import QtWidgets
+    import PySide2.QtCore as QtCore
+    from editor_python_test_tools.utils import TestHelper as helper
+    from editor_python_test_tools.utils import Report
+    import azlmbr.legacy.general as general
+    import pyside_utils
+    import editor_python_test_tools.EditorQtContainer as qtContainer
+    import editor_python_test_tools.scripting_tools_qt as scripting_tools_qt
+    from scripting_utils.scripting_constants import (WAIT_TIME_3, SCRIPT_CANVAS_UI, VARIABLE_TYPES,
+                                                     GRAPH_VARIABLES_QT)
+
+    general.idle_enable(True)
+
+    # 1) Open Script Canvas window and initialize the qt SC objects
+    general.open_pane(SCRIPT_CANVAS_UI)
+    helper.wait_for_condition(lambda: general.is_pane_visible(SCRIPT_CANVAS_UI), WAIT_TIME_3)
+    qtObjects = scripting_tools_qt.initialize_script_qt_canvas_objects()
+
+    # 3) Create Graph
+    scripting_tools_qt.create_new_sc_graph(qtObjects.sc_editor_main_window)
+
+    # 4) Create and verify the new variable exists in variable manager
+    scripting_tools_qt.create_new_variable(qtObjects, VARIABLE_TYPES[0])
```

Comment:
Can we wrap this behavior to be more explicit? Reading this I don't know what type of variable I'm creating. I'd love to be able to see this be more explicit and less mystery meaty. Maybe `VARIABLE_TYPES` can be a class to support dot operators so we can do stuff like `scvariabletype.Integer`.

Also, I'm thinking it would be better to rename `scripting_tools_qt` to `scirpt_canvs_editor_qt` and when imported as something like `QtSCEditor` or something easier to reference while coding.

Lastly, it would be nice if we didn't have to pass qtObjects into this helper or any of the other helpers.

Reply:
Refactored this so there's no parameters being passed in and it reads a bit more like pseudocode